### DISCUSSION TITLE
Signal inverted using USB Serial?

### DISCUSSION THREAD

**Discussion by flohoff** [fredlcore/BSB-LAN#486] [General]
Hi,
i put together the Level Converter and before digging into ESP32/Arduino code i thought i have a peek at the message telegrams on the BSB. So i connected a USB TTL 3.3v Serial and i got the telegrams. The issue now is that i see them inverted e.g. bitflipped.

```
0000 : #u........s#...  23 75 FF F4 F9 C2 F2 FA E6 B0 73 23 7F F5 F1 F8 
0010 : .........        F2 C2 FA E6 FF F3 E1 F6 E6 
```

Flipping the bits with ^0xff seems to produce more expected input:

```
0000 : .....=...O...... DC 8A 00 0B 06 3D 0D 05 19 4F 8C DC 80 0A 0E 07 
0010 : .=......0        0D 3D 05 19 00 0C 17 98 30 
```

I was a little puzzled to i had a look at the BSB-LAN code and i could not find any bit flipping in the RX path. I could find it in the TX path.

Is this expected or some electrical issue? What am i missing?

Flo

**Comment by fredlcore** [comment]
Yes, this is expected.

**Comment by flohoff** [comment]
So is inverting in BSB-LAN done in Software and where did i overlook it? Or is that and inverting in the ESP32 uart missing?

**Comment by fredlcore** [comment]
Both. 
Please understand that we are doing this as a hobby and taking care of users' problems already takes a lot of time already. If you want to re-invent the wheel, feel free to go ahead, but I don't have the time to explain all sorts of details that can be found on the web already, especially in our FHEM forum thread.
